development/build/transforms/remove-fenced-code.js

```diff
@@ -0,0 +1,425 @@
+        splicingIndices.push(indices[0]);
+      }
+    } else {
+      if (terminus !== DirectiveTerminuses.END) {
+        throw new Error(
+          getInvalidFencePairMessage(
+            filePath,
+            line,
+            `The second directive of a pair must be an "END" directive.`,
+          ),
+        );
+      }
+
+      /* istanbul ignore next: impossible until there's more than one command */
+      if (command !== currentCommand) {
+        throw new Error(
+          getInvalidFencePairMessage(
+            filePath,
+            line,
+            `Expected "END" directive to have command "${currentCommand}" but found "${command}".`,
+          ),
+        );
+      }
+
+      // Forbid empty fences
+      const { line: previousLine, indices: previousIndices } = parsedDirectives[
+        i - 1
+      ];
+      if (fileContent.substring(previousIndices[1], indices[0]).trim() === '') {
+        throw new Error(
+          `Empty fence found in file "${filePath}":\n${previousLine}\n${line}\n`,
+        );
+      }
+
+      if (shouldSplice) {
+        // Add end index of END directive line to splicing indices
+        splicingIndices.push(indices[1]);
+      }
+    }
+  }
+
+  // This indicates that the present build type should include all fenced code,
+  // and so we just returned the unmodified file contents.
+  if (splicingIndices.length === 0) {
+    return [fileContent, false];
+  }
+
+  /* istanbul ignore next: should be impossible */
+  if (splicingIndices.length % 2 !== 0) {
+    throw new Error(
+      `Internal error while transforming file "${filePath}":\nCollected an uneven number of splicing indices: "${splicingIndices.length}"`,
+    );
+  }
+
+  return [multiSplice(fileContent, splicingIndices), true];
+}
+
+/**
+ * Returns a copy of the given string, without the character ranges specified
+ * by the splicing indices array.
+ *
+ * The splicing indices must be a non-empty, even-length array of non-negative
+ * integers, specifying the character ranges to remove from the given string, as
+ * follows:
+ *
+ * `[ start, end, start, end, start, end, ... ]`
+ *
+ * @param {string} toSplice - The string to splice.
+ * @param {number[]} splicingIndices - Indices to splice at.
+ * @returns {string} The spliced string.
+ */
+function multiSplice(toSplice, splicingIndices) {
+  const retainedSubstrings = [];
+
+  // Get the first part to be included
+  // The substring() call returns an empty string if splicingIndices[0] is 0,
+  // which is exactly what we want in that case.
+  retainedSubstrings.push(toSplice.substring(0, splicingIndices[0]));
+
+  // This loop gets us all parts of the string that should be retained, except
+  // the first and the last.
+  // It iterates over all "end" indices of the array except the last one, and
+  // pushes the substring between each "end" index and the next "begin" index
+  // the the array of retained substrings.
```

Comment:
Typo:
```suggestion
  // t the array of retained substrings.
```

Reply:
Or `onto`? :thinking: 
```suggestion
  // onto the array of retained substrings.
```